resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/events/AutoBuildJob.java

```diff
@@ -111,6 +112,17 @@ synchronized void build(boolean needsBuild) {
 					}
 				}
 				break;
+			case RUNNING:
+				// In rare cases, we can end up in a situation where some other thread calls
+				// WorkManager.setBuild(true) while the AutoBuildJob is already running.
+				// In this case the AutoBuildJob should be rescheduled so we are certain that
+				// the new build request is eventually processed and does not get lost.
+				// Therefore, if a build is needed and the autobuild is enabled, then reschedule
+				// the job
+				if (isAutoBuilding && buildNeeded && !avoidBuild && Job.getJobManager().currentJob() != this) {
```

Comment:
it feels wrong that the conditions here are different to that in line 93. at least the logging in line 94 is probably wrong.

Reply:
I will update logging in an extra commit